src/widgets/TooltipWidget.cpp

```diff
@@ -49,123 +41,260 @@ TooltipWidget::TooltipWidget(BaseWidget *parent)
 
     auto windows = getApp()->windows;
     this->connections_.managedConnect(windows->gifRepaintRequested, [this] {
-        if (this->image_ && this->image_->animated())
+        for (int i = 0; i < this->visibleEntries_; ++i)
         {
-            this->refreshPixmap();
+            auto entry = this->entryAt(i);
+            if (entry && entry->animated())
+            {
+                entry->refreshPixmap();
+            }
         }
     });
 
     this->connections_.managedConnect(windows->miscUpdate, [this] {
-        if (this->image_ && this->attemptRefresh)
+        for (int i = 0; i < this->visibleEntries_; ++i)
         {
-            if (this->refreshPixmap())
+            auto entry = this->entryAt(i);
+            if (entry->getImage())
             {
-                this->attemptRefresh = false;
-                this->adjustSize();
+                if (entry->refreshPixmap())
```

Comment:
miscUpdate ticks every 100ms, so preferably refreshPixmap should be called only as long it returns false for given entry

Reply:
Good point, updated in [a275d67](https://github.com/Chatterino/chatterino2/pull/4314/commits/a275d67ba6a8d9c6ba6ffbb1817712bb5d110942)